src/stable.rs

```diff
@@ -591,10 +591,32 @@ impl fmt::Display for Literal {
     }
 }
 
-named!(token_stream -> ::TokenStream, map!(
-    many0!(token_tree),
-    |trees| ::TokenStream::_new(TokenStream { inner: trees })
-));
+fn token_stream(mut input: Cursor) -> PResult<::TokenStream> {
+    let mut trees = Vec::new();
+    loop {
+        let input_no_ws = skip_whitespace(input);
+        if input_no_ws.rest.len() == 0 {
+            break
+        }
+        if let Ok((a, comment)) = doc_comment(input_no_ws) {
+            input = a;
+            trees.push(TokenTree::Op(Op::new('#', Spacing::Alone)));
```

Comment:
If this is an inner doc comment `//!` or `/*!`, then we need both <kbd>#</kbd> <kbd>!</kbd>. See https://github.com/rust-lang/rust/issues/49656.